modules/4_nodes/nodes.tf

```diff
@@ -191,6 +202,18 @@ resource "openstack_compute_instance_v2" "worker" {
     }
 }
 
+resource "openstack_blockstorage_volume_v2" "worker" {
+  count       = var.worker["data_volume_count"] * var.worker["count"]
+  name        = "${var.cluster_id}-worker-${count.index}-volume"
+  size        = var.worker["data_volume_size"]
+}
+
+resource "openstack_compute_volume_attach_v2" "worker" {
+  count        = var.worker["data_volume_count"] * var.worker["count"]
+  instance_id  = openstack_compute_instance_v2.worker.*.id[floor(count.index / var.worker["data_volume_count"])]
+  volume_id    = element(openstack_blockstorage_volume_v2.worker.*.id, count.index)
```

Comment:
Please use the classic index based syntax eg: list[index]. This fails if the list is empty which could be held true for worker.count=0 and is recommended.

https://www.terraform.io/docs/language/functions/element.html